src/engine/Service.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Hannah Bast (bast@cs.uni-freiburg.de)
+
+#pragma once
+
+#include "engine/Operation.h"
+#include "engine/Values.h"
+#include "parser/ParsedQuery.h"
+
+// Class implementing the semantics of a SERVICE clause.
+//
+// TODO: Work in progress. For now, creating an empty result. Next step: use the
+// `Values` class to create a fixed result. Next step after that: use libcurl
+// to get the result from the remote endpoint and use the `Values` class to
+// parse that result.
+class Service : public Operation {
+ private:
+  // For each column, its multiplicity. Note that this can be computed
+  // precisely.
+  // TODO: For now, filled with random values.
+  std::vector<size_t> multiplicities_;
```

Comment:
We can only implement them if we materialize and analyze the full result during the query planning of which I am not sure that it's the right way to do this.

Reply:
What speaks against it? Do we not always have to execute the SERVICE query?